src/Optimizer/Transformer/OptimizeViewport.php

```diff
@@ -0,0 +1,131 @@
+ *
+ * This transformer will:
+ * * default to 'width=device-width' when the viewport is missing, which is the bare minimum that AMP requires;
+ * * extract properties from multiple viewport tags and merge them into a single tag;
+ * * remove the initial-scale=1 attribute if applicable to avoid unnecessary tap delay.
+ *
+ * @package ampproject/amp-toolbox
+ */
+final class OptimizeViewport implements Transformer
+{
+
+    /**
+     * Viewport settings to use for AMP markup.
+     *
+     * @var string
+     */
+    const AMP_VIEWPORT = 'width=device-width';
+
+    /**
+     * Xpath query to fetch the viewport meta tags.
+     *
+     * This transformer does not make use of the `Dom\Document::$viewport` helper, as it needs to
+     * deal properly with multiple viewport tags as well.
+     *
+     * @var string
+     */
+    const XPATH_QUERY = './/meta[@name="viewport"]';
+
+    /**
+     * Configuration store to use.
+     *
+     * @var TransformerConfiguration
+     */
+    private $configuration;
+
+    /**
+     * Instantiate a MinifyHtml object.
+     *
+     * @param TransformerConfiguration $configuration Configuration store to use.
+     */
+    public function __construct(TransformerConfiguration $configuration)
+    {
+        $this->configuration = $configuration;
+    }
+
+    /**
+     * Apply transformations to the provided DOM document.
+     *
+     * @param Document        $document DOM document to apply the transformations to.
+     * @param ErrorCollection $errors   Collection of errors that are collected during transformation.
+     */
+    public function transform(Document $document, ErrorCollection $errors)
+    {
+        $viewport = '';
+
+        $metaTags = $document->xpath->query(self::XPATH_QUERY);
+
+        if (! $metaTags->length) {
+            $viewport = self::AMP_VIEWPORT;
+        } else {
+            // Merge one or more meta[name=viewport] tags into one.
+            $parsedRules = [];
+
+            foreach ($metaTags as $metaTag) {
+                $propertyPairs = explode(',', $metaTag->getAttribute('content'));
+
+                foreach ($propertyPairs as $propertyPair) {
+                    $explodedPair = explode('=', $propertyPair, 2);
+                    if (isset($explodedPair[1])) {
+                        $parsedRules[ trim($explodedPair[0]) ] = trim($explodedPair[1]);
+                    }
+                }
+
+                $metaTag->parentNode->removeChild($metaTag);
+            }
+
+            // Remove initial-scale=1 to leave just width=device-width in order to avoid a tap delay that hurts FID.
+            if (
+                $this->configuration->get(OptimizeViewportConfiguration::REMOVE_INITIAL_SCALE_VIEWPORT_PROPERTY)
+                && isset($parsedRules[Attribute::INITIAL_SCALE])
+                && abs((float) $parsedRules[Attribute::INITIAL_SCALE] - 1.0) < 0.0001
+            ) {
+                unset($parsedRules[Attribute::INITIAL_SCALE]);
```

Comment:
The attribute is not `initial-scale`, the attribute is `viewport`. `initial-scale` is one common _value_ of the attribute `viewport`. Because of this, I suggested using `Attribute::VIEWPORT_INITIAL_SCALE` here, to match with what we already have for some other attributes where we store common _values_ for the attribute.

See https://github.com/ampproject/amp-toolbox-php/blob/0.7.0/src/Attribute.php#L985-L1136

Reply:
There is no `VIEWPORT_INITIAL_SCALE` constant in the `Attribute` interface. Beside we are not looking for `viewport` here. `$parsedRules` is generated from `content="width=device-width,initial-scale=1"` part of the viewport meta tag.

Reply:
You're right, attribute value would be `initial-scale=1`, not `initial-scale`. What we need is a "viewport property"...

Ok, revert that change for now, then, and let's go with what you had initially.

Reply:
The current code is actually same as before. I'm just using the constant now instead of string `initial-scale`.

Reply:
Yes, but the `initial-scale` is not an attribute, so it doesn't belong into the `Attribute` interface. It's better to use an internal constant then.

Reply:
Updated. Added the constant in `OptimizeViewport` class and used that instead of the constant from `Attribute`.